Build our docker images on top of gpuci/miniconda-cuda

ohadmo opened the pull request:
This images are based on `gpuci/miniconda-cuda`  repository instead of the deprecated `gpuci/rapidsai-base` & `gpuci/rapidsai-base-driver` repositories.

Fixes https://github.com/clara-parabricks/GenomeWorks/issues/460

ohadmo committed:
Build our docker images on top of gpuci/miniconda-cuda

Instead of the oudated repositories  gpuci/rapidsai-base & gpuci/rapidsai-base-driver

ohadmo committed:
Change `gpuci/miniconda-cuda` images to be based on `nvidia/cuda-devel`

ohadmo committed:
Add Dockerfile for centOS based on `gpuci/miniconda-cuda`

tijyojwad commented on Dockerfile:
```diff
@@ -8,23 +8,89 @@
-        wget \
-        libssl-dev      # VariantWorks `cyvcf2` dependency
+# Capture argument used for FROM
+ARG CC_VERSION
```
I see `ARG CC_VERSION` and `ARG PYTHON_VERSION` near line 14 as well. Can those be reused here?

ohadmo replied:
You're right, no need to declare these at all before the `FROM` command.

tijyojwad commented on Dockerfile:
```diff
@@ -8,23 +8,89 @@
 
 # Install htslib
-RUN wget https://github.com/samtools/htslib/releases/download/1.9/htslib-1.9.tar.bz2 && tar xvf htslib-1.9.tar.bz2 && cd htslib-1.9 && ./configure && make -j16 install
+RUN wget https://github.com/samtools/htslib/releases/download/1.9/htslib-1.9.tar.bz2 \
```
we're about to add `htslib` as a submodule into GenomeWorks, so I think we can skip installing this from source. We could add the `apt` install for this instead. I think the package name is `libhts-dev`

ohadmo replied:
Alright, I made this change for Ubuntu. For centOS I think this would be better to build it from source since it requires downloading 3 rpm's from https://centos.pkgs.org/7/epel-x86_64/htslib-devel-1.9-5.el7.x86_64.rpm.html

tijyojwad replied:
hmm then it might be better to revert it, or maybe even install through `conda`. best to have the same version across ubuntu and centos to avoid version related bugs

ohadmo replied:
Done. Moved the installation of `htslib` to Anaconda for both

tijyojwad replied:
awesome, thank you!

tijyojwad commented on Dockerfile:
```diff
@@ -8,23 +8,89 @@
+# Create rapids conda env and make default
+RUN source activate base \
+    && conda install -y gpuci-tools \
+    && gpuci_conda_retry create --no-default-packages --override-channels -n rapids \
```
does it have to be called rapids, or can be called `parabricks`?

ohadmo replied:
No problem, I'm gonna keep the symlink from 'gdf' to 'parabricks'  env in line 90, until we update this across all our repositories

tijyojwad replied:
Perfect, thanks!

tijyojwad commented on Dockerfile.drivers:
```diff
@@ -8,23 +8,23 @@
+ARG DRIVER_VER="440"
+
+# Update and add pkgs
+RUN apt-get update -q \
```
just curious why this one needs the `headless` components whereas the other ones don't?

ohadmo replied:
There are actually two ways to build the driver images, we can either be based on the image we just created in `gpuci/clara-genomics-base` (`./Dockerfile`). or we can be based on `gpuci/miniconda-cuda-driver` (which already contains this logic) and copy the content of  `./Dockerfile`, I went with the former.

tijyojwad replied:
Yep I think that's totally fine. I was just curious why this particular image needed this `headless` thing. Maybe that's a question more for @raydouglass

ohadmo replied:
I see these commands were updated in https://github.com/rapidsai/gpuci-build-environment/commit/4d4337f44f2f90c420ae1e323d5fd97674ca75d3

tijyojwad replied:
but that change shows those dependencies were removed?

ohadmo replied:
@tijyojwad I changed the driver image to be built based on `gpuci/miniconda-cuda-driver` and not `gpuci/clara-genomics-base` since different CUDA versions require a different driver installation as in https://gpuci.gpuopenanalytics.com/job/gpuci/job/docker/job/miniconda-cuda-driver/
so we don't need to reconfigure https://gpuci.gpuopenanalytics.com/job/clara-genomics/job/docker/job/clara-genomics-base-driver/ for now

tijyojwad replied:
Awesome, thanks!

ohadmo committed:
Build our images based on `gpuci/miniconda-cuda`- fixed review comments

Changed conda environment name to 'parabricks'

Move htslib installation to Anaconda

ohadmo committed:
Install cudatoolkit through conda

ohadmo committed:
build driver images on top of gpuci/miniconda-cuda-driver
